File: 0141_Linked List Cycle/LinkedListCycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        var p  = head;
        var set = new HashSet<ListNode>();
        while(p!=null){
            if(set.Contains(p)) return true;
            set.Add(p);
            p = p.next;
        }
        return false;
    }
}

File: 0141_Linked List Cycle/LinkedListCycle.java
public class Solution {
    public boolean hasCycle(ListNode head) {
        if(head == null || head.next == null) return false;
        ListNode f = head;
        ListNode s = head;
        
        while(f != null && f.next != null) {
            f = f.next.next;
            s = s.next;
            
            if(f == s) return true;
        }
        
        return false;
    }
}